libethash-cuda/CUDAMiner.cpp

```diff
@@ -527,7 +523,7 @@ void CUDAMiner::search(
 			if (found_count)
 				for (uint32_t i = 0; i < found_count; i++)
 					if (s_noeval)
-						farm.submitProof(Solution{nonces[i], *((const h256 *)mixes[i]), w, m_new_work});
+						farm.submitProof(Solution{nonces[i], mixes[i], w, m_new_work});
```

Comment:
The "*((const h256 *)mixes[i])" fails to to achieve what I would expect is a h256 pointer to the 8 word mix, under latest GCC.

Reply:
No undefined behavior! It's always the case that a reference to an array by name is equivalent to a pointer to its 1st element. Since mixes is 2 dimensional, the mixes[j] refers to an array by name!